test/dynamo/test_higher_order_ops.py

```diff
@@ -1063,7 +1068,7 @@ def true_fn(x):
                 def false_fn(x):
                     return x.sum() - z.sum() - self.buffer.sum()
 
-                return control_flow.cond(y, true_fn, false_fn, [x])
+                return control_flow.cond(x, true_fn, false_fn, [y])
```

Comment:
Why did this change? (does it actually matter?)

Reply:
Yes, here is a typo where x is tensor[True/False] but y is an interger tensor. Previsouly, we can work for integer tensors. But now, we add an assertion that it must be boolean tensors. This allows us to catch this typo and correct it.